### ISSUE TITLE
Vagrantfile is outdated / not working

### ISSUE THREAD

**Issue by Zyigh** [apache/openwhisk#4468]
## Environment details:
Local deployement with vagrant 2.2.4 on Mac OS Mojave (10.14.4) and VirtualBox 5.2.2

## Steps to reproduce the issue:

Go on `tools/vagrant` and execute `./hello` (which is basically a `vagrant up`)

## Provide the expected results and outputs:

This should create a virtual machine with everything installed and set up properly

## Provide the actual results and outputs:

3 problems are preventing the construction of the virtual machine : 
* Java Installer is outdated.
* Grub installer asks for interaction
* npm is missing, so installation won't work

## Additional information you deem important:

I couldn't test it on different OS / version. Also, the grub interraction problem appeared today on my computer. The [solution I found here](https://github.com/hashicorp/vagrant/issues/289#issuecomment-12408200) is from 2013.

I had to modify the Vagrantfile to make it work : 

I removed the line `su vagrant -c 'source all.sh oracle'` and replaced it with 
```bash
su vagrant -c 'source all.sh'
curl -sL https://deb.nodesource.com/setup_8.x | sudo bash - && apt-get install -y nodejs
sudo DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" dist-upgrade
```

Removing the option "oracle" to all.sh install scripts that are still supported by ubuntu.
After that line I install node 
And then run the script found to prevent interraction with grub during vagrant up

Also I thought it would be important to write this as an issue as I had to search many different places to find a solution so this can help, but I'm not sure it would be a good idea to create a pull request as I couldn't test it in many environment (such as different VirtualBox or vagrant versions)

**Comment by rabbah** [comment]
thanks @Zyigh for the issue and suggested fix.

**Comment by emilyaherbert** [comment]
This fix worked for me with vagrant 2.2.5, virtual box 6.0.12, and Ubuntu 18.04.

**Comment by rabbah** [comment]
We're not maintaining this image actively and I will propose removing it on the Apache dev list.
